### DISCUSSION TITLE
cpg0010-caie-drugresponse

### DISCUSSION THREAD

**Discussion by ErinWeisbart** [broadinstitute/cellpainting-gallery#68] [Dataset descriptions]
Update as generated:  
Imaging Platform project tag = YYYY_MM_DD_ProjectName
[Link to profile repo]  
[Link to publication repo]  

- [ ] Metadata collection form filled out by collaborator or Imaging Platform member (https://airtable.com/shrVxz9DcoMlDoCBI)
- [ ] Metadata completely filled out in Project Profiler Database (Imaging Platform internal use only)
- [x] Segmentation/Feature extraction complete and in cellpainting-gallery
- [x] Profiling complete and in cellpainting-gallery

Project completion:
- [ ] Run validation script to ensure data is completely uploaded (https://github.com/jump-cellpainting/data-validation)
- [ ] Update cellpainting-gallery/README.md to add dataset to README

If data is being published, prepare for publication:
- [ ] Run Distributed-BioFormats2Raw to create .ome.zarr files
- [ ] Upload (meta)data to IDR (images remain hosted in cellpainting-gallery).

Once published:
- [ ] Make IDR entry public
- [ ] Update cellpainting-gallery/README.md and open-data-registry/cellpainting-gallery.yml with publication

**Comment by GaiaGerbaka** [comment]
In the context of trying to harmonize the gallery for BFF, I'm attempting to add information from the publication to the metadata. However, I'm unable to link the different cell lines (Ovcar3 (ovarian), MiaPaCa2 (pancreatic), and MCF7 cells wild-type and mutant for p53 (breast)) to the plates (Week<number>_<identifier>). It would be helpful to provide a link between plates and cells since this study presents pharmacologic response across four cancer cell types

**Reply by ErinWeisbart** [reply]
@bethac07 Do you have any insight into this?

The [BBBC021 page](https://bbbc.broadinstitute.org/BBBC021) is titled "Human MCF7 cells" and says "The images are of MCF-7 breast cancer cells treated for 24 h with a collection of 113 small molecules at eight concentrations. The cells were fixed, labeled for DNA, F-actin, and Β-tubulin, and imaged by fluorescent microscopy as described [[Caie et al. Molecular Cancer Therapeutics, 2010](http://dx.doi.org/10.1158/1535-7163.MCT-09-1148)]."  but the Caie et al manuscript lists the 4 cell lines mentioned above. 

The metadata linked in the BBBC page doesn't say anything about cell line anywhere.

**Reply by bethac07** [reply]
I suspect the BBBC021 might be just the MCF7 data, and perhaps the other line data was not shared? I suspect this mainly because the math (13,200 sites total representing 904 compound-times-dose-points with 4 sites per well == 3.65 wells per compound-dose-point (and they say some were missing, so this means ~~quadruplicate wells)). Unless a LOT of replicates are missing, it's not enough data for 4 cell lines. So then the question becomes - is the data in the CPG just the 55 plates in the BBBC, or is it additional plates as well?

All of this is best-guesstimate, because BBBC021 was before my time - we'd probably need to go into the dumps of the old wiki to dig further, which we CAN, but is not my preference per se.

**Reply by ErinWeisbart** [reply]
I believe this was copied straight from BBBC to CPG so I think we'll proceed with best-guess that it's just MCF7 cells. Thanks!

**Comment by ErinWeisbart** [comment]
Notes from [dataset page on BBBC](https://bbbc.broadinstitute.org/BBBC021)

A subset of the compound-concentrations have been identified as clearly having one of 12 different primary mechanisms of action. Mechanistic classes were selected so as to represent a wide cross-section of cellular morphological phenotypes. The differences between phenotypes in some cases were very subtle: we identified 6 of the 12 mechanisms visually (Actin disruptors, Aurora kinase inhibitors, Eg5 inhibitors, Microtubule destabilizers, Microtubule stabilizers, and Epithelial); the remainder were defined based on the literature.

All compounds were tested at eight doses. The top concentration was different for many of the compounds and was carefully selected from the literature. Not all concentrations are available in this dataset. Missing concentrations are due to one of three factors:

The dose was determined to be inactive. Activity was defined by setting a threshold on the Mahalanobis distance from the set of DMSO profiles: profiles of doses that were outside this threshold were considered active. The feature space corresponded to measurements extracted by a proprietary software tool used at AstraZeneca.
The dose was determined to be overly toxic, i.e., the images had no cells or very few cells.
The images did not pass QC, that is, they were either out of focus wells or contained image artifacts.

**Comment by ErinWeisbart** [comment]
I parsed the metadata in the [BBBC021 page](https://bbbc.broadinstitute.org/BBBC021) to match CPG formatting with:

```python
import pandas as pd
import os

df = pd.read_csv('/Users/eweisbar/Downloads/BBBC021_v1_image.csv')
df['plate_map_name'] = df['Image_PathName_DAPI'].str.split('/',expand=True)[1]
cols = ['Replicate','Image_Metadata_Compound','Image_Metadata_Concentration','plate_map_name','Image_Metadata_Well_DAPI']
df = df[cols]
df.columns = [x.replace('Image_Metadata_','').replace('Well_DAPI','well_position') for x in df.columns]
df = df.drop_duplicates()
# move cols to front
col_data = df.pop('well_position')
df.insert(0, 'well_position', col_data)
col_data = df.pop('plate_map_name')
df.insert(0, 'plate_map_name', col_data)
# make platemap.txts
for platemap in df['plate_map_name'].unique():
    week = platemap.split('_')[0]
    
    if not os.path.exists(f'/Users/eweisbar/Desktop/metadata/platemaps/{week}/platemap/'):
        os.makedirs(f'/Users/eweisbar/Desktop/metadata/platemaps/{week}/platemap/',exist_ok=True)
    df.loc[df['plate_map_name']==platemap].to_csv(f'/Users/eweisbar/Desktop/metadata/platemaps/{week}/platemap/{platemap}.txt', sep='\t', index=False)

    # make barcode_platemaps.csv
    dfmap = pd.DataFrame(columns=['Assay_Plate_Barcode','Plate_Map_Name'])
    dfmap['Assay_Plate_Barcode'] = dfmap['Plate_Map_Name'] = [x for x in df['plate_map_name'].unique() if week in x]
    dfmap.to_csv(f'/Users/eweisbar/Desktop/metadata/platemaps/{week}/barcode_platemap.csv', index=False)

# make external metadata

moa = pd.read_csv('/Users/eweisbar/Downloads/BBBC021_v1_moa.csv')
print (moa.shape)
compound = pd.read_csv('/Users/eweisbar/Downloads/BBBC021_v1_compound.csv')
print (compound.shape)

df = moa.merge(compound, on='compound', how='outer')
print(df.shape)
if not os.path.exists('/Users/eweisbar/Desktop/metadata/external_metadata/'):
    os.makedirs('/Users/eweisbar/Desktop/metadata/external_metadata/', exist_ok=True)
df.to_csv('/Users/eweisbar/Desktop/metadata/external_metadata/external_metadata.csv', index=False)
```
